src/js/components/pages/greendash/CompareCard.jsx

```diff
@@ -181,71 +188,255 @@ const CampaignCard = ({baseFilters}) => {
+						},
+					},
+				},
+			},
+		},
+	};
 
-	const subcard = (mode === 'quarter') ? (
-		<QuartersCard {...props} />
-	) : (
-		<CampaignCard {...props} />
-	);
+	return <NewChartWidget type="bar" {...chartProps} />;
+};
 
-	return <GreenCard title="How do your ad emissions compare?" className="carbon-compare">
-		<div className="d-flex justify-content-around mb-2">
-			<ButtonGroup>
-				<ModeButton name="quarter" mode={mode} setMode={setMode}>
-					Quarter
-				</ModeButton>
-				<ModeButton name="campaign" mode={mode} setMode={setMode} disabled={campaignModeDisabled} >
-					Campaign
-				</ModeButton>
-			</ButtonGroup>
-		</div>
-		{subcard}
-		<GreenCardAbout>
-			<p>Explanation of quarterly and per-campaign emissions comparisons</p>
-		</GreenCardAbout>
-	</GreenCard>;
+/**
+ * @param {string} str
+ * @return {string}
+ */
+const capitalizeFirstLetter = (str) => {
+	if (!str || typeof str !== "string") {
+		return "";
+	}
+	const lower = str.toLowerCase();
+	return lower.charAt(0).toUpperCase() + lower.slice(1);
+};
+
+/**
+ * Get the map of impressions per Country. Filter out data that have less than 0.1% of total impressions.
+ * @param {GreenBuckets | null} buckets
+ * @param {Number | null} allCount
+ * @returns {Object.<string, number> | null}
+ */
+const getCountryMapFiltered = (buckets, allCount, filterThreshold = 0.001) => {
+	if (!buckets || !allCount) return null;
+
+	const threshold = filterThreshold * allCount;
+
+	const countryMap = buckets.reduce((acc, val) => {
+		if (val.count < threshold) {
+			return acc;
+		}
+
+		if (!Object.prototype.hasOwnProperty.call(acc, val.key)) {
+			acc[val.key] = val.count;
+		} else {
+			acc[val.key] += val.count;
+		}
+		return acc;
+	}, {});
+
+	return countryMap;
+};
+
+/**
+ * @param {Object} obj
+ * @param {string} obj.mode
+ * @param {GreenBuckets} obj.buckets
+ * @param {BaseFilters} obj.props
+ * @returns {JSX.Element}
+ */
+const SwitchCard = ({ mode, buckets, props }) => {
+	switch (mode) {
+		case "quarter":
+			return <QuartersCard {...props} />;
+		case "campaign":
+			return <CampaignCard {...props} />;
+		case "benchmarks":
+			return <BenchmarksCard buckets={buckets} benchmarksData={props?.benchmarksData} />;
+		default:
+			return <QuartersCard {...props} />;
+	}
+};
+
+const CompareCard = ({ dataValue, ...props }) => {
+	const [mode, setMode] = useState("quarter");
+	const [benchmarksData, setBenchmarksData] = useState();
+
+	const per1000 = isPer1000();
+
+	const BenchmarksDataPv = useMemo(async () => {
+		if (!per1000) return null;
+
+		const countryData = await getCarbon({ ...props.baseFilters, breakdown: ['country{"count":"sum"}'] }).promise;
+		const countryMap = getCountryMapFiltered(countryData.by_country?.buckets, countryData.allCount);
+
+		const urlParams = getUrlVars();
+		/** TODO: Upgrade to last quarter later */
+		const period = getPeriodFromUrlParams({ ...urlParams, period: "last-month" });
+
+		const countryCo2pm = await Promise.all(
```

Comment:
Hello @wongws11 
Good work. Could you add some more comments to this patch of code?
We're getting country data to do a weighted sum. Let's make that clear with comments. Thanks!

Reply:
No problem. Will do.